libplatsupport/src/mach/spacemit/timer.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2025, 10xEngineers
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <platsupport/mach/timer.h>
+#include <stdint.h>
+#include <utils/util.h>
+
+#define MAX_TIMEOUT_NS (SPACEMIT_TIMER_MAX_TICKS * (NS_IN_S / SPACEMIT_TIMER_TICKS_PER_SECOND))
+
+static void spacemit_timer_enable_clocks(spacemit_timer_t *timer)
+{
+    volatile uint32_t *base   = (volatile uint32_t *)((uint8_t *)timer->vaddr + 0x1000); // APB_CLK_BASE
+    uint32_t offset           = APBC_TIMERx_CLK_RST_OFFSET;
+    uint32_t clk_select_shift = 0x4;
+    uint32_t init_value       = 0x0;
+
+    init_value |= (1 << 0);
+	init_value |= (1 << 1);
+    init_value |= K1_TCCR_CS0_VALUE << clk_select_shift;
+
+	*(base + offset/4) = init_value;
+}
+
+void spacemit_timer_enable(spacemit_timer_t *timer)
+{
+    assert(timer);
+    assert(timer->regs);
+    assert(timer->timer_n < SPACEMIT_NUM_TIMERS);
+
+    timer->regs->tcer  |= (1u << timer->timer_n);
+}
+
+
+void spacemit_timer_disable(spacemit_timer_t *timer)
+{
+    assert(timer);
+	assert(timer->regs);
+
+	timer->regs->tcer  &= ~(1u << timer->timer_n);
+}
+
+uint64_t spacemit_timer_get_time(spacemit_timer_t *timer)
+{
+    assert(timer);
+	assert(timer->regs);
+
+	uint8_t n = timer->timer_n;
+	/* the timer value counts down from the load value */
+	uint64_t value_l = (uint64_t)(SPACEMIT_TIMER_MAX_TICKS - (timer->regs->tcnt[n]));
+	uint64_t value_h = (uint64_t)timer->value_h;
+
+	/* Include unhandled interrupt in value_h */
+	uint32_t status = timer->regs->tsr[n];
+	if (status & 0x1) {
+		value_h += 1;
+	}
+
+	uint64_t value_ticks = (value_h << 32) | value_l;
+
+	/* convert from ticks to nanoseconds */
+	uint64_t value_whole_seconds = value_ticks / SPACEMIT_TIMER_TICKS_PER_SECOND;
+	uint64_t value_subsecond_ticks = value_ticks % SPACEMIT_TIMER_TICKS_PER_SECOND;
+	uint64_t value_subsecond_ns =
+		(value_subsecond_ticks * NS_IN_S) / SPACEMIT_TIMER_TICKS_PER_SECOND;
+	uint64_t value_ns = value_whole_seconds * NS_IN_S + value_subsecond_ns;
```

Comment:
```suggestion
	uint64_t whole_seconds = value_ticks / SPACEMIT_TIMER_TICKS_PER_SECOND;
	uint64_t subsecond_ticks = value_ticks % SPACEMIT_TIMER_TICKS_PER_SECOND;
	uint64_t subsecond_ns = (subsecond_ticks * NS_IN_S) / SPACEMIT_TIMER_TICKS_PER_SECOND;
	uint64_t value_ns = whole_seconds * NS_IN_S + subsecond_ns;
```
Prefixing everything with "value_" just makes everything harder to read.